src/CachePeers.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_CACHEPEERS_H_
+#define SQUID_CACHEPEERS_H_
+
+#include "CachePeer.h"
+#include "mem/PoolingAllocator.h"
+
+#include <memory>
+#include <vector>
+
+/// cache_peer configuration storage
+class CachePeers
+{
+public:
+    using CachePeerList = std::vector< std::unique_ptr<CachePeer>, PoolingAllocator< std::unique_ptr<CachePeer> > >;
```

Comment:
To avoid a false implication that this is a list (or any other specific type):

```suggestion
    using Storage = std::vector< std::unique_ptr<CachePeer>, PoolingAllocator< std::unique_ptr<CachePeer> > >;
```